Branch-SDK/Branch-SDK/ContentDiscoverer.m

```diff
@@ -0,0 +1,287 @@
+                    rootView = [[rootView subviews] objectAtIndex:subviewId];
+                } else {
+                    foundView =false;
+                    break;
+                }
+            }
+        }
+        if (foundView == true) {
+            NSString *contentVal = [self getContentText:rootView];
+            if (contentVal) {
+                viewTxt = contentVal;
+            }
+        }
+    }
+    return viewTxt;
+}
+
+- (UIViewController *)getActiveViewController {
+    UIViewController *rootViewController = [UIApplication sharedApplication].keyWindow.rootViewController;
+    return [self getActiveViewController:rootViewController];
+    
+}
+
+- (UIViewController *)getActiveViewController:(UIViewController *)rootViewController {
+    UIViewController *activeController;
+    if ([rootViewController isKindOfClass:[UINavigationController class]]) {
+        activeController = ((UINavigationController *)rootViewController).topViewController;
+    }
+    else if ([rootViewController isKindOfClass:[UITabBarController class]]) {
+        activeController = ((UITabBarController *)rootViewController).selectedViewController;
+    }
+    else {
+        activeController = rootViewController;
+    }
+    return activeController;
+}
+
+- (void)addFormatedContentData:(NSMutableArray *)contentDataArray withText:(NSString *)contentData clearText:(BOOL)isClearText {
+    if (contentData != nil && contentData.length > cdManifest.maxTextLen) {
+        contentData = [contentData substringToIndex:cdManifest.maxTextLen];
+    }
+    if (!isClearText) {
+        contentData = [self hashContent:contentData];
+    }
+    [contentDataArray addObject:contentData];
+}
+
+- (NSString*)hashContent:(NSString *)content {
+    const char *ptr = [content UTF8String];
+    unsigned char md5Buffer[CC_MD5_DIGEST_LENGTH];
+    CC_MD5(ptr, (CC_LONG)strlen(ptr), md5Buffer);
+    NSMutableString *output = [NSMutableString stringWithCapacity:CC_MD5_DIGEST_LENGTH * 2];
+    for (int i = 0; i < CC_MD5_DIGEST_LENGTH; i++) {
+        [output appendFormat:@"%02x",md5Buffer[i]];
+    }
+    return output;
+}
+
+- (NSString *)getContentText:(UIView *)view {
+    NSString * contentData = nil;
+    if ([view respondsToSelector:@selector(text)]) {
+        contentData = [view performSelector:@selector(text) withObject:nil];
+    }
+    if (contentData == nil || contentData.length == 0) {
+        if ([view respondsToSelector:@selector(attributedText)]) {
+            contentData = [view performSelector:@selector(attributedText) withObject:nil];
+        }
+    }
+    
+    if (contentData == nil || contentData.length == 0) {
+        if ([view isKindOfClass:UIButton.class]) {
+            contentData = [view performSelector:@selector(titleLabel) withObject:nil];
+        }
+        else if ([view isKindOfClass:UITextField.class]) {
+            contentData = [view performSelector:@selector(placeholder) withObject:nil];
```

Comment:
I'd hesitate to grab the placeholder text since this is likely to be hardcoded and generic. We should probably ignore this one.